app/models/dto.py

```
@@ -61,6 +64,46 @@ class AvailabilityRequest(BaseModel):
+    @model_validator(mode='after')
+    def validate_time_range(self) -> 'AvailabilityRequest':
+        start_h = int(self.start_hour.split(':')[0])
+        end_h = int(self.end_hour.split(':')[0])
+        if start_h >= end_h:
+            raise ValueError(f"종료 시간({self.end_hour})은 시작 시간({self.start_hour})보다 최소 1시간 이후여야 합니다.")
+        
+        # 날짜 및 과거/미래 제한 로직 검증 (KST 기준)
+        try:
+            req_date = datetime.strptime(self.date, "%Y-%m-%d").date()
+        except ValueError:
+            raise ValueError(f"날짜 형식이 올바르지 않습니다. (YYYY-MM-DD 포맷 필요, 입력값: {self.date})")
```

Ruff (0.15.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)